src/main/java/duke/parser/Parser.java

```diff
@@ -78,6 +78,19 @@ private static void eventCheck(String[] tasks, String userInput) throws DukeExce
         }
     }
 
+    /**
+     * This method checks to ensure that the user provides
+     * a keyword to search for from the task list.
+     *
+     * @param tasks The user's input split into an array of strings.
+     * @throws DukeException If there is no keyword to search for.
+     */
+    private static void findCheck(String[] tasks) throws DukeException {
+        if (tasks.length <= 1) {
+            throw new DukeException("OOPS!!! Please tell me what you want to find!!");
+        }
+    }
```

Comment:
Quite interesting that you have a separate methods to check the validity of a command, as opposed to checking the validity within your FindTaskCommand/AddCommand and throwing your DukeException there. I didn't ever encounter a need for this approach, I'm curious is this method works better or more reliably?